### DISCUSSION TITLE
How to initialize each connection in a pool after create

### DISCUSSION THREAD

**Discussion by ArminEberle** [sequelize/sequelize#15746] [Help & Questions] [answered]
Hi there,

I'm just migrating to sequelize from our own 'Multi-Database-Driver' framework code, hoping to get rid of most of our own stuff :-)

Right now I'm facing an issue: 

The mysql pool driver allows us to run queries on a connection right after the connection was created for the pool. 
We use this to do some settings on the connection:

* timezone for example, hence I found the timezone setting in the options which seems fine
* `SET @@group_concat_max_len=50000;` which sets the allowed max result length for a group_concat aggregate function. I can't find a setting for this in the constructor options. 

This is of course MySQL specific code and I'm doing this only when connecting to the MySQL/Maria DB.

So I'm wondering if there is a way to initialize every connection after creation and before it is used in the pool for the 'normal' queries?

**Comment by ephys** [comment, marked as answer]
The `afterConnect` hook runs right after a connection has been established, it's dispatched on the sequelize instance

Sequelize 7 documentation: https://sequelize.org/docs/v7/other-topics/hooks/#instance-sequelize-hooks
Sequelize 6 documentation: https://sequelize.org/docs/v6/other-topics/hooks/#connection-hooks

**Reply by ArminEberle** [reply]
Thank you very much, that works for me!
